File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/entity/TeamEntity.kt
package dev.tpavani.octopoints.db.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(tableName = "teams", foreignKeys = [
    ForeignKey(
        entity = MatchEntity::class,
        parentColumns = ["id"],
        childColumns = ["matchId"],
        onDelete = ForeignKey.CASCADE
    )
])
class TeamEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Int,
    val matchId: Int,
    val name: String,
    var total: Int,
    var partial: Int,
)

File: octopoints_flutter/lib/db/entity/TeamEntity.dart
import 'package:floor/floor.dart';
import 'package:octopoints_flutter/db/entity/MatchEntity.dart';

@Entity(tableName: 'teams', foreignKeys: [
  ForeignKey(
    childColumns: ['matchId'],
    parentColumns: ['id'],
    entity: MatchEntity,
    onDelete: ForeignKeyAction.cascade,
  )
])
class TeamEntity {
  @PrimaryKey(autoGenerate: true)
  final int? id;
  final int matchId;
  final int partial;
  final int total;
  final int status;

  TeamEntity(
    this.id,
    this.matchId,
    this.partial,
    this.total,
    this.status
  );
}
